autoload/lsp.vim

```diff
@@ -214,7 +217,30 @@ function! s:call_did_save(buf, server_name, result, cb) abort
 endfunction
 
 function! s:on_text_document_did_close() abort
-    call lsp#log('s:on_text_document_did_close()', bufnr('%'))
+    let l:buf = bufnr('%')
+    call lsp#log('s:on_text_document_did_close()', l:buf)
+endfunction
+
+function! lsp#get_last_file_content(server_name, buf) abort
+    if has_key(s:file_content, a:buf) && has_key(s:file_content[a:buf], a:server_name)
+        return s:file_content[a:buf][a:server_name]
+    endif
+    return []
+endfunction
+
+function! lsp#update_file_content(server_name, buf, new) abort
```

Comment:
should we make this private for now and rename it to `s:update_file_content`?

Reply:
`lsp#send_request` automatically flushes the current buffer to external consumers of that api doesn't need to worry about manual flushing.